src/engine/world/world.h

```diff
@@ -292,6 +292,15 @@ class World final
 
   void addPickupWidget(Sprite sprite, size_t count)
   {
+    auto isSameItemAndSameAction = [sprite](ui::PickupWidget item) -> bool {
+      return item.m_sprite.textureId == sprite.textureId && item.m_duration == 75_frame;
+    };
+    if(auto it = std::find_if(m_pickupWidgets.begin(), m_pickupWidgets.end(), isSameItemAndSameAction); 
+       it != std::end(m_pickupWidgets))
```

Comment:
```suggestion
       it != m_pickupWidgets.end())
```
That's inconsistent to how you access the end iterator in the line above.